### ISSUE TITLE
Java support

### ISSUE THREAD

**Issue by fdionisi** [zed-industries/zed#5301]
**Describe the solution you'd like**
Would be great to have Java support, at least syntax highlight. LSP integration is a (very) nice to have.

**Comment by SomeoneToIgnore** [comment]
Java is somewhat special due to a very deep project file trees, which are working bad with current project panel.
Hence, before adding another language into the mix and causing extra influx of issues, we should rather start with improving the project panel.

See https://github.com/zed-industries/zed/pull/6935#issuecomment-1913640078 for more details.

**Comment by wmstack** [comment]
Hopefully the Java support here doesn't suffer from the bugs that are found in VSCode Java, such as this https://github.com/redhat-developer/vscode-java/issues/1886

Many things I want to say. First of all, context menu pollution is not appreciated. There has to be a way to limit what is in the context menu instead of allowing every language to pollute it with silly things like "Add/Remove folder from Java Source Path".

Not sure what the cause of the specific bug here, is, but I have many Java Projects in a single directory and don't want them to interfere with each other running.

**Comment by masterscode** [comment]
I really like the performance and memory efficiency of Zed but Java is my 9-5; it'll be nice if a good support of java is added

**Comment by wmstack** [comment]
100% and I think this is key for driving adoption. The main features I care about are support for auto-imports, a fast project creation pathway and the ability to run/debug projects with a click of a button

Some things I don't want to see in Java for Zed are things like https://github.com/redhat-developer/vscode-java/issues/1886

**Comment by stephensheridan** [comment]
Yes, please add Java support. The lack of Java support is the main reason I have not totally switched over.

**Comment by shbfy** [comment]
+1

**Comment by Surechun** [comment]
Please add Java support.>_<

**Comment by pedrosmmc** [comment]
Any news regarding Java's support availability?

**Comment by AydenHex** [comment]
Some news ?

**Comment by JosephTLyons** [comment]
There is an extension in the store right now - it looks to only provide syntax highlighting, but mentions that it will be adding language server support soon.

<img width="1187" alt="SCR-20240412-lhwb" src="https://github.com/zed-industries/zed/assets/19867440/3297b8dd-8c48-4d65-9c3c-4cc3db351209">
